src/main/java/command/CommandSort.java

```diff
@@ -45,28 +45,24 @@ public CommandRes execute() {
             System.out.println("Sorry, there are no expenses tracked currently.");
             System.out.println(MESSAGE_DIVIDER);
         } else {
-            if (!sortBy.equals("C") && !sortBy.equals("D")) {
-                System.out.println("Please indicate you want your expenses sorted by Date/Category? ");
-                System.out.println("Enter: D (represent Date); C (represent Category)");
-                assert false;
-                return null;
-            }
-
-            System.out.println(MESSAGE_DIVIDER_LIST);
-            if (sortBy.equals("C")) {
+            switch(sortBy) {
+            case "C":
+                System.out.println(MESSAGE_DIVIDER_SORTEDLIST);
                 sortByCategory();
-                for (int i = 0; i < expenseListCategory.size(); i++) {
-                    System.out.print((i + 1) + ".");
-                    System.out.println(expenseListCategory.get(i).sortedDisplay(sortBy));
-                }
-            } else {
+                displayByCategory();
+                System.out.println(MESSAGE_DIVIDER);
+                break;
+            case "D":
+                System.out.println(MESSAGE_DIVIDER_SORTEDLIST);
                 sortByDate();
-                for (int i = 0; i < expenseListDate.size(); i++) {
-                    System.out.print((i + 1) + ".");
-                    System.out.println(expenseListDate.get(i).sortedDisplay(sortBy));
-                }
+                displayByDate();
+                System.out.println(MESSAGE_DIVIDER);
+                break;
+            default:
+                System.out.println("Please indicate you want your expenses sorted by Date/Category? ");
+                System.out.println("Enter: D (represent Date); C (represent Category)");
```

Comment:
Could be potentially confusing. The message after keying in `sort` seems to suggest that subsequent input can be C or D only.
![Screenshot 2023-04-05 at 18 52 31 1](https://user-images.githubusercontent.com/79389472/230060217-f7b690f6-512f-4237-bf5e-34b7c04b9ccf.png)